File: 1/twoSum.py
"""
1. use a dict to store the num and index
2. search the array, find (target - num) in dict
time O(n)
"""
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        
        for i,num in enumerate(nums):
            if target-num in d:
                return d[target-num]+1,i+1
            
            d[num] = i
        return -1,-1

File: 1/twoSum.c
/**
 * calculate all the two sum in array
 * return the sum which equals to target
 * the time O(n^2)
 */
int* twoSum(int* nums, int numsSize, int target) {
    int *ret = (int*)malloc(sizeof(int)*2);
    
    for(int i = 0; i < numsSize; i++){
        for(int j = i + 1; j < numsSize; j++){
            int sum = nums[i] + nums[j];
            if(sum == target){
                ret[0] = i + 1;
                ret[1] = j + 1;
                return ret;
            }
        }
    }
    return NULL;
}